src/scaling.jl

```diff
@@ -0,0 +1,192 @@
+"""
+function apply!(
+    A::AbstractArray, scaling::MeanStdScaling;
+    dims=:, inverse=false, eps=1e-3, kwargs...
+)
+    if isempty(scaling.mean) || isempty(scaling.std)
+        # Populate with mean and std of this data, once and for all
+        populate_stats!(scaling, A; dims=dims)
+    end
+
+    dims == Colon() && return _apply!(A, scaling; inverse=inverse, eps=eps, kwargs...)
+
+    for (i, x) in enumerate(eachslice(A; dims=dims))
+        _apply!(x, scaling; name=Symbol(i), inverse=inverse, eps=eps, kwargs...)
+    end
+
+    return A
+end
+
+"""
+    _apply!(
+        A::AbstractArray, t::MeanStdScaling;
+        dims=:, inverse=false, kwargs...
+    )
+
+Applies [`MeanStdScaling`](@ref) to each element of `A`.
+Optionally specify the `dims` to apply [`MeanStdScaling`](@ref) along certain dimensions,
+and `inverse=true` to reconstruct the originally scaled data from `A`.
+For `inverse=false`, all 0 std values get replaced by the value `eps` before scaling.
+"""
+function _apply!(
+    A::AbstractArray, scaling::MeanStdScaling;
+    name=nothing, inverse=false, eps=1e-3, kwargs...
+)
+    name = name === nothing ? :all : name
+    μ = scaling.mean[name]
+    σ = scaling.std[name]
+    if inverse
+        A[:] = μ .+ σ .* A
+    else
+        # Avoid division by 0
+        # If std is 0 then data was uniform, so the scaled value would end up ≈0
+        # Therefore the particular `eps` value should not matter much.
+        σ_safe = σ == 0 ? eps : σ
+        A[:] = (A .- μ) ./ σ_safe
+    end
+    return A
+end
+
+function apply(A::AbstractArray, scaling::MeanStdScaling; kwargs...)
+    return apply!(_try_copy(A), scaling; kwargs...)
+end
+
+"""
+    apply!(table::T, scaling::MeanStdScaling; cols=nothing, kwargs...)::T where T
+
+Applies the [`MeanStdScaling`](@ref) to each of the specified columns in the `table`.
+If no `cols` are specified, then the [`MeanStdScaling`](@ref) is applied to all columns.
+Optionally specify `inverse=true` to reconstruct the originally scaled data.
+For `inverse=false`, all 0 std values get replaced by the value `eps` before scaling.
+"""
+function apply!(table::T, scaling::MeanStdScaling; cols=nothing, kwargs...)::T where T
+    # TODO: We could probably handle iterators of tables here
+    Tables.istable(table) || throw(MethodError(apply!, (table, scaling)))
+
+    if isempty(scaling.mean) || isempty(scaling.std)
+        # Populate with mean and std of this data, once and for all
+        populate_stats!(scaling, table; cols=cols)
+    end
+
+    # Extract a columns iterator that we should be able to use to mutate the data.
+    # NOTE: Mutation is not guaranteed for all table types, but it avoid copying the data
+    columntable = Tables.columns(table)
+
+    cnames = cols === nothing ? propertynames(columntable) : cols
+    for cname in cnames
+        apply!(getproperty(columntable, cname), scaling; name=cname, kwargs...)
```

Comment:
@nicoleepp the `name` keyword here and in the array version prevents re-using the general apply methods. It makes sure the correct mean/std is applied to the correct column. Any ideas?

My ideas:
- Require users to keep the order consistent (should be fine for arrays but annoying for tables)
- Add "iter kwargs" to general methods, which get iterated and passed to the inner `_apply!` one-by-one.
- Still have a custom apply method, but it constructs a new MeanStdScaling with the stats in the correct order, then passes that version of the transform to the general method.

Reply:
I would also need to add `kwargs...` to the general apply for tables, in order to pass on `inverse` and `eps` arguments.

Reply:
> Can the general methods just have the name keyword without affecting other transforms

Seems fine to me

> I would also need to add kwargs... to the general apply! for tables, in order to pass on inverse and eps arguments. I think that should be there anyway.

I agree